### ISSUE TITLE
Cannot get rewrite_tag filter to work as expected

### ISSUE THREAD

**Issue by mowings** [fluent/fluent-bit#3589]
## Bug Report

NOTE: ~~It is highly probably that I am doing something stupid~~ EDIT: No, this actually looks like a bug. tl;dr a "bare wildcard" (`*`) does not seem to work as expected with the `rewrite_tag` filter.

**Describe the bug**
I have a simple set of data, which I pass through a rewrite_tag filter. When "keep" (in the rule) is set to "true" two records are output as expected -- one with the old tag, and one wth the new. But if keep is set to false, no record are emitted.

I would expect that the newlly tagged record would be emitted

**To Reproduce**
Start with the following config file:

```
[SERVICE]
    flush        1
    log_level    info

[INPUT]
    Name dummy
    Dummy {"type": "foo-bar", "color": "blue", "label": {"name": null}}

[FILTER]
    Name                 rewrite_tag
    Match                *
    Rule                 $type ^foo-bar foo.bar true

[OUTPUT]
    name        stdout
    match       *
```

As expected, both records are logged with the exected tags:
```
[0] dummy.0: [1622661841.128193340, {"type"=>"foo-bar", "color"=>"blue", "label"=>{"name"=>nil}}]
[0] foo.bar: [1622661841.128193340, {"type"=>"foo-bar", "color"=>"blue", "label"=>{"name"=>nil}}]
```

But if the config file is changed so that the keep parameter is false in the rule, nothing is emitted at all
```
[SERVICE]
    flush        1
    log_level    info

[INPUT]
    Name dummy
    Dummy {"type": "foo-bar", "color": "blue", "label": {"name": null}}

[FILTER]
    Name                 rewrite_tag
    Match                *
    Rule                 $type ^foo-bar foo.bar false

[OUTPUT]
    name        stdout
    match       *
```

In this case, fluentbit shows no output at all

**Expected behavior**

I would expect fluentbit in the second case to simply output:

```
[0] foo.bar: [1622661841.128193340, {"type"=>"foo-bar", "color"=>"blue", "label"=>{"name"=>nil}}]
```
Instead the entire record seems to get eaten.

**Your Environment**

* Version used: 1.7.4
* Configuration: fluentbit is simply run with the above config file at the command line
* Server type and version: linux EC2 instance
* Operating System and version: Ubunti 18.04
* Filters and plugins: As above

**Additional context**

I am simply attempting to either throttle or redirect certain types of log messages. I am CERTAIN this is my fault -- but I cannot see my error. Any help would be greatly appreciated

EDIT:

So this DOES seem to be a bug.  The following works:

```
[SERVICE]
    flush        1
    log_level    info

[INPUT]
    Name dummy
    Dummy {"type": "foo-bar", "color": "blue", "label": {"name": null}}

[FILTER]
    Name                 rewrite_tag
    Match                dum*
    Rule                 $type ^foo-bar foo.bar false

[OUTPUT]
    name        stdout
    match       *
```

But changing the Match rule in the filter to be a lone `*` does not:

```
[SERVICE]
    flush        1
    log_level    info

[INPUT]
    Name dummy
    Dummy {"type": "foo-bar", "color": "blue", "label": {"name": null}}

[FILTER]
    Name                 rewrite_tag
    Match                *
    Rule                 $type ^foo-bar foo.bar false

[OUTPUT]
    name        stdout
    match       *
```

Basically it looks like the match in `rewrite_tag` does not support a bare `*` wildcard. Not a huge deal, but this seems like a bug. `Match *` seems to be supported elsewhere.

**Comment by nokute78** [comment]
I think this is not a bug.
In this case, infinite loop occurred (following step 2-5)
We should not use `Match *` for rewrite_tag.

1. dummy generates event `dummy.0: [1622661841.128193340, {"type"=>"foo-bar", "color"=>"blue", "label"=>{"name"=>nil}}]`
2. rewrite_tag rewrites tag and event will be `foo.bar: [1622661841.128193340, {"type"=>"foo-bar", "color"=>"blue", "label"=>{"name"=>nil}}]`
3. rewrite_tag emits the new event and doesn't pass to out_stdout since `Rule false`
4. rewrite_tag also modifies the new event because `match *` and the new event also contains `"type"=>"foo-bar"`
5. back to 2

**Comment by nokute78** [comment]
Note: #2570
